jvb/src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -170,19 +168,18 @@
     private List<AdaptiveSourceProjection> adaptiveSourceProjections
             = Collections.emptyList();
 
-    /**
-     * The default video constraints to use for endpoints without video
-     * constraints.
-     */
-    private static final VideoConstraints defaultVideoConstraints
-        = new VideoConstraints(BitrateControllerConfig.thumbnailMaxHeightPx());
-
     /**
      * The map of endpoint id to video constraints that contains the video
      * constraints to respect when allocating bandwidth for a specific endpoint.
      */
     private ImmutableMap<String, VideoConstraints> videoConstraintsMap = ImmutableMap.of();
 
+    /**
+     * A modified copy of the original video constraints map, augmented with video constraints for the endpoints that
+     * fall outside of the last-n set + endpoints not announced in the videoConstraintsMap.
+     */
+    private Map<String, VideoConstraints> effectiveConstraintsMap = Collections.EMPTY_MAP;
```

Comment:
Sure, so all the client sends is who's selected and max resolution MAXp. In my past work I pass these messages through a compatibility layer to compute the video constraints: for every selected endpoint it creates a video constraints instance of max resolution MAXp (the actual computation is a tiny bit more complicated than that because the compat. layer also takes into account tile-view). The video constraints computed by the compatibility layer end-up in `videoConstraintsMap`.

Before the bitrate allocation algorithm runs, we need video constraints for all endpoints, i.e. contraints for the thumbnails (video constraints of 180p) and for the people that are outside of last-n (video constraints of 0p). These are the _effective_ video constraints.